plugins/modules/ndo_pod_profile.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Akini Ross (@akinross) <akinross@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_pod_profile
+short_description: Manage Pod Profiles on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage Pod Profiles on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.2 (NDO v4.4) and later.
+author:
+- Akini Ross (@akinross)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a Fabric Resource Policy template.
+    - This parameter or O(template_id) is required.
+    type: str
+  template_id:
+    description:
+    - The ID of the template.
+    - The template must be a Fabric Resource Policy template.
+    - This parameter or O(template) is required.
+    type: str
+  name:
+    description:
+    - The name of the Pod Profile.
+    type: str
+    aliases: [ pod_profile ]
+  uuid:
+    description:
+    - The UUID of the Pod Profile.
+    - This parameter is required when the O(name) needs to be updated.
+    type: str
+    aliases: [ pod_profile_uuid ]
+  description:
+    description:
+    - The description of the Pod Profile.
+    type: str
+  pods:
+    description:
+    - A list of pod IDs to be used in the Pod Profile.
+    - When provided, the Pod Profile will apply to pods specified in the range.
+    - When an empty list O(pods=[]) is provided, the Pod Profile will apply to all pods.
+    - When not provided on inital create, the Pod Profile will apply to all pods.
+    type: list
+    elements: int
+    aliases: [ blocks ]
+  pod_settings:
```

Comment:
We have previously been nesting this reference objects into dictionaries. 
ie.
```
pod_settings: dictionary {
   name: str, required,
   template: str,
   template_id: str
}
pod_settings_uuid: str
```
Have we decided to format specs without nesting now?

Reply:
I do not know all the constant changes that have decided but if you can provide me this then I can change. I thought at least it was like this in the past that for required attributes we would flatten the structure, because the user input would be easier.

Reply:
Thanks for changing it. I believe this was decided in a standup or PR conversation before we improved our process to document these types of decisions shared in the stand-up channel. 
Some of the most recent module in ansible-mso are doing this now. eg [(1)](https://github.com/CiscoDevNet/ansible-mso/pull/632/files#diff-972541f30f207ab1daddbdf780f43a453bbff2e70c7b753e99840782583f64d5R71) [(2)](https://github.com/CiscoDevNet/ansible-mso/pull/656/files#diff-6c3da0f6aab6fe2dfd4d8115ada8847a743d15a8acf9923b71d22d6c1846d5cdR74) [(3)](https://github.com/CiscoDevNet/ansible-mso/pull/639/files#diff-0ad43095b91619161276e948dbc4ef0c260c82991ab9b82e1250ba9974a72345R79)